src/main/java/sirius/db/mixing/properties/AmountProperty.java

```diff
@@ -60,7 +60,7 @@ public void create(EntityDescriptor descriptor,
             AmountProperty amountProperty = new AmountProperty(descriptor, accessPath, field);
 
             try {
-                if (field.get(descriptor.getType().getDeclaredConstructor().newInstance()) == null) {
+                if (field.get(accessPath.apply(descriptor.getType().getDeclaredConstructor().newInstance())) == null) {
```

Comment:
add a testcase to reproduce this bug / ensure that it is fixed? :)

Reply:
Not sure how to tap into that part of the code in a test case